src/qt/bitcoin.cpp

```diff
@@ -188,24 +212,85 @@ int main(int argc, char *argv[])
 
     SetupEnvironment();
 
-    // Set default value to exit properly. Exit code 42 will trigger restart of the wallet.
-    int currentExitCode = 0;
- 
+    // Do this early as we don't want to bother initializing if we are just calling IPC
+    ipcScanRelay(argc, argv);
+
+    // Command-line options take precedence:
+    // Before this would of been done in main then config file loaded.
+    // We will load config file here as well.
+    ParseParameters(argc, argv);
+
+    ReadConfigFile(mapArgs, mapMultiArgs);
+
+    // Here we do it if it was started with the snapshot argument and we not TestNet
+    if (mapArgs.count("-snapshotdownload") && !mapArgs.count("-testnet"))
+    {
+        Upgrade Snapshot;
+
+        // Let's check make sure gridcoin is not already running in the data directory.
+        if (!LockDirectory(GetDataDir(), ".lock", false))
+        {
+            fprintf(stderr, "Cannot obtain a lock on data directory %s.  Gridcoin is probably already running.", GetDataDir().string().c_str());
+
+            exit(1);
+        }
+
+        else
+            Snapshot.SnapshotMain();
+
+        // Delete snapshot regardless of result.
+        Snapshot.DeleteSnapshot();
+    }
+
+    /** Start Qt as normal before it was moved into this function **/
+    StartGridcoinQt(argc, argv);
+
+    // We got a request to apply snapshot from GUI Menu selection
+    // We got this request and everything should be shutdown now.
+    // Except what we cannot shutdown.
+    // In future once code base for the databases are updated we won't need
+    // the wallet user to start the QT wallet themselves after the snapshot
+    // has been applied. We will be able to restart it ourselves
+    // See Bitcoin's database files and pointers involved and Reset()
+    // What prevents us from starting the wallet again internally
+    // is the lingering of database.
+    // The linger does not affect the snapshot process but prevents restart of wallet within main area
+
+    /** This is only if the GUI menu option was used! **/
+    if (fSnapshotRequest)
+    {
+        UpgradeQt test;
+
+        if (test.SnapshotMain())
```

Comment:
We're not deleting the snapshot.zip file at all when the user starts the snapshot process from the GUI menu item.

Reply:
willl look at this

Reply:
this the one u are referring to